### PR TITLE
Task7

### PR DESCRIPTION
**Pull request by pawkli95**
Updated Gradle and extracted versions of dependencies.
Added profiles and logging.

### CODE REVIEW

File: app/src/main/java/pl/futurecollars/invoicing/config/DatabaseConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+@Configuration
+public class DatabaseConfiguration {
+
+    @ConditionalOnProperty(name = "invoicing-system.database", havingValue = "file")
+    @Bean
+    public Database getFileBasedDatabase(@Value("${invoicing-system.invoicesFile}") String invoicesFile,
+                                         @Value("${invoicing-system.idsFile}") String idsFile) {
+        FileService invoices = new FileService(invoicesFile);
+        FileService ids = new FileService(idsFile);
+        log.info("Created file database");
```
Comment (radoslawdabrowski) [inline app/src/main/java/pl/futurecollars/invoicing/config/DatabaseConfiguration.java:25]: These kind of logs should be `debug`

File: app/src/main/java/pl/futurecollars/invoicing/config/DatabaseConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        return new FileBasedDatabase(new JsonService<Invoice>(), invoices, ids);
+    }
+
+    @Bean
+    @ConditionalOnProperty(name = "invoicing-system.database", havingValue = "inmemory")
+    public Database getInMemoryDatabase() {
+        log.info("Created inmemory database");
```
Comment (radoslawdabrowski) [inline app/src/main/java/pl/futurecollars/invoicing/config/DatabaseConfiguration.java:32]: as previous

File: app/src/main/java/pl/futurecollars/invoicing/config/DatabaseConfiguration.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        return new InMemoryDatabase();
+    }
+
+    @Bean
+    @ConditionalOnProperty(name = "invoicing-system.database", havingValue = "fileTest")
+    public Database getTestFileBasedDatabase(@Value("${invoicing-system.invoicesFile}") String invoicesFile,
+                                                  @Value("${invoicing-system.idsFile}") String idsFile) {
+        FileService invoices = new FileService(invoicesFile);
+        FileService ids = new FileService(idsFile);
+        log.info("Created test file database");
```
Comment (radoslawdabrowski) [inline app/src/main/java/pl/futurecollars/invoicing/config/DatabaseConfiguration.java:42]: as previous

File: app/src/main/java/pl/futurecollars/invoicing/config/DatabaseConfiguration.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+        return new FileBasedDatabase(new JsonService<Invoice>(), invoices, ids);
+    }
+
+    @Bean
+    @ConditionalOnProperty(name = "invoicing-system.database", havingValue = "inmemoryTest")
+    public Database getTestInMemoryDatabase() {
+        log.info("Created test inmemory database");
```
Comment (radoslawdabrowski) [inline app/src/main/java/pl/futurecollars/invoicing/config/DatabaseConfiguration.java:49]: as previous

File: app/src/test/groovy/pl/futurecollars/invoicing/db/memory/InMemoryDatabaseTest.groovy [hunk 5]
```diff
@@ -7,6 +7,6 @@ class InMemoryDatabaseTest extends DatabaseTest{
 
     @Override
     Database getDatabase() {
-        return new InMemoryDatabase()
+        return new InMemoryDatabase();
```
Comment (radoslawdabrowski) [inline app/src/test/groovy/pl/futurecollars/invoicing/db/memory/InMemoryDatabaseTest.groovy:10]: unnecessary semicolon

**Review by radoslawdabrowski — APPROVED** [review]

**Review by katdunaj — APPROVED** [review]

**Pull request merged**